module/documents/activity/mixin.mjs

```diff
@@ -715,6 +720,29 @@ export default Base => class extends PseudoDocumentMixin(Base) {
     if ( config.consume === false ) return updates;
     const errors = [];
 
+    // Handle action economy
+    if ( ((config.consume === true) || config.consume.action) && (this.activation.type === "legendary") ) {
+      const containsLegendaryConsumption = this.consumption.targets
+        .find(t => (t.type === "attribute") && (t.target === "resources.legact.value"));
+      const count = this.activation.value ?? 1;
+      const legendary = this.actor.system.resources?.legact;
+      if ( legendary && !containsLegendaryConsumption ) {
+        let message;
+        if ( legendary.value === 0 ) message = "DND5E.ACTIVATION.Warning.NoActions";
+        else if ( count > legendary.value ) message = "DND5E.ACTIVATION.Warning.NotEnoughActions";
```

Comment:
Not sure there's much value in distinguishing between these two cases.

Reply:
Just used the same warning pattern as the other consumption bits, which distinguish. More useful for 2014 content which have things that spend more than 1 legendary action.